# RNA-seq: a step-by-step analysis pipeline

## Prerequisites/Dependencies to run a RNA-seq pipeline.

### JAVA version above 11.0

To install Java on a server using wget, you can use the following steps:

Open a terminal window on the server and login.
```
ssh <username>@<server_ip>
```
Use the wget command to download the Java installation file.
```
wget https://download.java.net/java/GA/jdk17/jdk-17_linux-x64_bin.tar.gz

```
Use the dpkg command to install the Java installation file.
```
sudo dpkg -i jdk-17_linux-x64_bin.tar.gz
```
Restart the server.
```
sudo reboot
```

### Miniconda (https://docs.conda.io/projects/miniconda/en/latest/)
```
mkdir -p ~/miniconda3
wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O ~/miniconda3/miniconda.sh
bash ~/miniconda3/miniconda.sh -b -u -p ~/miniconda3
rm -rf ~/miniconda3/miniconda.sh
```

# Data Analysis of RNA-Seq data
Programs required: it is recommended that the user has anaconda installed, through which all required programs can be installed. Assuming that anaconda is available, all the required programs can be installed using the following:

```
#Install the required programs using anaconda

conda install -c bioconda fastqc
conda install -c bioconda trimmomatic
conda install -c bioconda multiqc
conda install -c bioconda star
conda install -c bioconda samtools

#For differential expression using DESeq2
conda create DEseq2 r-essentials r-base

conda install DEseq2 -c bioconda bioconductor-deseq2 
conda install DEseq2 -c r r-ggplot2 
```

## Introduction
This pipeline is compatabile with RNA-seq reads generated by Illumina.

Before we run FastQC, you should be on a compute node in an interactive session. Please run the following srun command if you are not on a compute node. An interactive session is very useful to test tools and workflows.
```
srun --pty -t 0-6:00 --mem 5G /bin/bash
```

Now, after you are on interactive mode, develop the scripts required for analysis and submit the jobs using"
```
sbatch "filename.sh"
```

And, the squeue command is used to pull up information about jobs in the queue, by default this command will list the job ID, partition, username, job status, number of nodes, and name of nodes for all jobs queued or running within SLURM.
```
squeue -u username
```
And if you want to end the interactive mode type *exit*

### Quality Control on raw reads(fastqc)

The raw sequence data should first be assessed for quality. FastQC reports can be generated for all samples to assess sequence quality, GC content, duplication rates, length distribution, K-mer content and adapter contamination. For paired-end reads, run fastqc on both files, with the results output to the current directory:

```
fastqc *.fastq.gz -o /path/to/results/dir # Fastqc on 8 samples of RNA-Seq data
```

### Trimming(trimmomatic)

Trimming is a useful step of pre-alignment QC, which removes low quality reads and contaminating adapter sequences. Here, the tool *trimmomatic* is used to trim the data. For paired-end data:
run fastqc on all the rnaseq fastq files.....

This script is specific to run multiple samples at once, so a for loop is created

```
#!/bin/bash
#SBATCH --cpus-per-task=2 --mem-per-cpu=16g --ntasks=1
#SBATCH hetjob
#SBATCH --cpus-per-task=2 --mem-per-cpu=1g  --ntasks=8
#srun run.app              

mkdir -p ../path/to/output/folder                                                                                                                                                                                             
module load trimmomatic

for f in *_R1_001.fastq.gz # for each sample
do
  n=${f%%_R1_001.fastq.gz}
  java -jar /shared/centos7/anaconda3/2021.11/envs/BINF-12-2021/pkgs/trimmomatic0.39hdfd78af_2/share/trimmomatic-0.39-2/trimmomatic.jar PE -threads 40 $1 /path/to/input/FASTA_file_1/${n}_R1_001.fastq.gz/path/to/input/FASTA_file_2/${n}_R2_001.fastq.gz /path/to/output/trimmed_FASTA_file_1/${n}_R1_trimmed.fastq.gz /path/to/output/unpaired_FASTA_file_1/${n}_R1_unpaired.fastq.gz /path/to/output/trimmed_FASTA_file_2/${n}_R2_trimmed.fastq.gz /path/to/output/unpaired_FASTA_file_2/${n}_R2_unpaired.fastq.gz ILLUMINACLIP:/shared/centos7/anaconda3/2021.11/envs/BINF-12-2021/pkgs/trimmomatic-0.39-hdfd78af_2/share/trimmomatic-0.39-2/adapters/TruSeq3-PE.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36                                                                                                          done
```
### Building the STAR index(STAR)
The raw RNA-seq data in fastq format will be aligned to the reference genome, along with a reference transcriptome, to output two alignment files: 
the genome alignment and 
the transcriptome alignemnt.

The RNA reads are aligned using the STAR. The reference genome used is the GRCh38 assembly from ncbi. The genome can be downloaded using below commands.
This version of the recent GRCh38 reference genome excludes alternative contigs which may cause fragments to map in multiple locations. 
The downloaded genome should be indexed with STAR. The indexing also requires a file containing gene annotation, which comes in a gtf format. The user should aim to use the most up-to-date reference files, while ensuring that the format is the same as the reference genome. 

```
mkdir References # create a folder to dowload the reference genome
cd References

#downlaod the index and an annotation file
wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M25/GRCm38.primary_assembly.genome.fa.gz
wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M25/gencode.vM25.annotation.gtf.gz

#unzip the files using gunzip
gunzip GRCm38.primary_assembly.genome.fa.gz
gunzip gencode.vM25.annotation.gtf.gz

cd ..
mkdir index #create an index folder to save genome indices
genomedir : specify/path/to/genomeindex

module load star
                                                  
–runThreadN: number of threads
–runMode: genomeGenerate mode
–genomeDir: /path/to/store/genome_indices
–genomeFastaFiles: /path/to/FASTA_file
–sjdbGTFfile: /path/to/GTF_file
–sjdbOverhang: readlength -1
```

## Aligning reads to the genome(STAR)

STAR can then be run to align the fastq raw data to the genome. If the fastq files are in the compressed .gz format, the --readFilesCommand zcat argument is added. The output file should be unsorted, as required for the downstream quantification step using Salmon. The following options are shown according to the ENCODE recommendations.

For paired-end data:
```
# define variables
index = /path/to/indexfolder

# get our data files
FILES=/path/to/trimmed/files/*_trimmed.fastq.gz

module load star
for f in $FILES

do
  echo $f
  base=$(basename $f .fastq.gz)
  echo $base
  STAR --runThreadN 8 --genomeDir $index --readFilesIn $f --outSAMtype BAM SortedByCoordinate --quantMode GeneCounts --readFilesCommand zcat --outFileNamePrefix $base"_"                                                                                                                                                                                                                                                                                                           
done
```

