```{r}
# Install and load necessary packages
#install.packages("DESeq2")
#install.packages("ggplot2")
#install.packages("pheatmap")
#install.packages("openxlsx")
library(DESeq2)
library(ggplot2)
library(pheatmap)
library(openxlsx)

# Read and format the count data
counts <- read.csv("/Users/hp/Desktop/rnaseq/counts.txt", sep="\t", header=TRUE, skip=1, row.names = 1)
counts <- counts[,-c(1:5)]
counts$Ensemble_ID <- rownames(counts)
counts <- counts[, c(ncol(counts), 1:(ncol(counts)-1))]
colnames(counts) <- c("Ensemble_ID", "WT_rep1_R1","WT_rep1_R2", "Knockout_rep1_R1","Knockout_rep1_R2", "WT_rep2_R1", "WT_rep2_R2", "Knockout_rep2_R1", "Knockout_rep2_R2")

#counts <- counts[, c("Ensemble_ID", 1:(ncol(counts)-1))]

# Extract metadata
samples <- data.frame(row.names=colnames(counts)[-1], 
                      condition=c(rep("WT", 4), rep("Knockout", 4)))

# Create DESeq2 object
dds <- DESeqDataSetFromMatrix(countData = counts[,-1], colData = samples, design = ~ condition)

# Differential expression analysis
dds <- DESeq(dds)

# Extract results
res <- results(dds)

# Handle NAs in padj column
res$padj[is.na(res$padj)] <- 1

# Categorize genes and add Ensemble_ID column
res_df <- as.data.frame(res)
res_df$Ensemble_ID <- rownames(res_df)
res_df <- res_df[, c("Ensemble_ID", setdiff(names(res_df), "Ensemble_ID"))]

res_df$category <- ifelse(res_df$padj < 0.05 & res_df$log2FoldChange > 1, "Upregulated",
                          ifelse(res_df$padj < 0.05 & res_df$log2FoldChange < -1, "Downregulated", "Non-significant"))

# Select top 20 significant upregulated genes based on adjusted p-value
filtered_genes <- res_df[res_df$padj < 0.05 & res_df$log2FoldChange > 1,]
top20 <- filtered_genes %>% arrange(desc(log2FoldChange)) %>% head(20)
#top20 <- head(filtered_genes[order(-filtered_genes$log2FoldChange)], 20)

# Create an Excel workbook
wb <- createWorkbook()

# Add the two sheets with the respective data
addWorksheet(wb, "DEG")
writeData(wb, "DEG", res_df)

addWorksheet(wb, "Top 20 Upregulated DEG")
#writeData(wb, "Top 20 Upregulated DEG", top20)
writeData(wb, "Top 20 Upregulated DEG", top20)

# Save the workbook
saveWorkbook(wb, "/Users/hp/Desktop/rnaseq/DEGs_Upregulated.xlsx", overwrite = TRUE)

# Visualization: Volcano plot
volcano_plot <- ggplot(res_df, aes(x=log2FoldChange, y=-log10(padj), color=category)) +
  geom_point(size=1.5) +
  theme_minimal() +
  labs(title="Volcano plot", x="log2_Fold_Change", y="-log10 adjusted p-value") +
  scale_color_manual(values=c("Upregulated"="red", "Downregulated"="blue", "Non-significant"="grey50"))

volcano_plot

# Visualization: Heatmap
# Extracting top 20 genes based on p-value
topgenes <- rownames(top20)
mat <- assay(dds)[topgenes,]
mat <- log2(mat + 1)
pheatmap(mat, show_rownames=TRUE)

# Return top 20 upregulated genes
top20
```